CMakeLists.txt

```diff
@@ -231,13 +231,13 @@ endif ()
 
 include(StaticAnalyzers)
 if (OPENASSETIO_ENABLE_CLANG_FORMAT)
-    enable_clang_format()
+    SET_CLANG_FORMAT()
```

Comment:
Renamed these macros to conform with the style, but also not cause confusion with `ENABLE_CLANG_FORMAT` variables

Reply:
If we're renaming anyway, we should namespace them, i.e. prefix with `OPENASSETIO_`.

Reply:
Ended up setting these back to what they were before, so leaving them as is.